### PR TITLE
Imp/gh37 implement indication of default state on parameter value sets in ed browser and product tree; fixes #37

### LINKED ISSUE #37: Implement indication of default state on parameter value sets in ED browser and product tree
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of the CDP4-IME
- [X] I have searched [open](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

When a ParameterValueSet is associated to the default state it should be shown in bold in both the ElementDefinition tree and the Product Tree 
(Note: aligning with finite states browser, where this is applied already).

### Logs

  - [ ] There are no logs regarding the issue
  - { ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- CDP4 version:
  - [ ] CDP4-IME: 
  - [ ] Other: 
  -
- Web Services:
  - [ ] CDP4 Web Services version: 
  - [ ] OCDT WSP version: 
- Environment (Operating system, version and so on):
  - [ ] Microsoft Excel (32/64 bit): 
  - [ ] Windows OS (32/64 bit): 
  - [ ] Additional information:

### PR DESCRIPTION
**Pull request by kamilatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
When a ParameterValueSet is associated to the default state it will show in bold in the ElementDefinition tree, Product Tree and EditParameterDialog.

### CODE REVIEW

File: EngineeringModel/ViewModels/Dialogs/Rows/ParameterValueBaseRowViewModel.cs [hunk 1]
```diff
@@ -179,6 +183,15 @@ public bool IsMultiSelect
             }
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether this <see cref="PossibleFiniteState"/> is the default value of the <see cref="PossibleFiniteStateList"/>
```
Comment (alexatrhea) [inline EngineeringModel/ViewModels/Dialogs/Rows/ParameterValueBaseRowViewModel.cs]: This should be:
```
Gets or sets a value indicating whether the <see cref="ActualFiniteState"/> associated with this row is the default value of the <see cref="PossibleFiniteStateList"/>
```

File: EngineeringModel/ViewModels/ElementDefinitionBrowser/ElementDefinitionTreeRows/ParameterStateRowViewModel.cs [hunk 2]
```diff
@@ -45,5 +51,14 @@ public ParameterStateRowViewModel(ParameterBase parameterBase, Option option, Ac
                 this.Disposables.Add(stateListener);
             }
         }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this <see cref="PossibleFiniteState"/> is the default value of the <see cref="PossibleFiniteStateList"/>
```
Comment (alexatrhea) [inline EngineeringModel/ViewModels/ElementDefinitionBrowser/ElementDefinitionTreeRows/ParameterStateRowViewModel.cs]: This should be:
```
Gets or sets a value indicating whether the <see cref="ActualFiniteState"/> associated with this row is the default value of the <see cref="PossibleFiniteStateList"/>
```

File: ProductTree/ViewModels/ProductTreeRows/ActualFiniteStateRowViewModel.cs [hunk 3]
```diff
@@ -56,6 +61,7 @@ public ActualFiniteStateRowViewModel(ActualFiniteState actualFiniteState, ISessi
             : base(actualFiniteState, session, containerViewModel)
         {
             this.IsPublishable = false;
+            this.IsDefault = actualFiniteState.IsDefault;
```
Comment (alexatrhea) [inline ProductTree/ViewModels/ProductTreeRows/ActualFiniteStateRowViewModel.cs]: This is not entirely correct I think. The row might not react if the state is no longer default. Please verify.

File: EngineeringModel/ViewModels/Dialogs/Rows/ParameterValueBaseRowViewModel.cs (hunk 1 above)
Comment (kamilatrhea) [reply EngineeringModel/ViewModels/Dialogs/Rows/ParameterValueBaseRowViewModel.cs]: OK

File: EngineeringModel/Views/Dialogs/ParameterSubscriptionDialog.xaml [hunk 4]
```diff
@@ -54,6 +54,7 @@
                 <lc:LayoutItem Label="Model Code: ">
                     <dxe:TextEdit Name="ModelCode" 
                                   Text="{Binding Path=ModelCode, Mode=OneWay, UpdateSourceTrigger=PropertyChanged}"
+                                  FontWeight="Bold"
```
Comment (lxatrhea) [inline EngineeringModel/Views/Dialogs/ParameterSubscriptionDialog.xaml]: Was this change intentional? If so, OK!

File: CDP4CommonView/Resources/ValueSetTemplates.xaml [hunk 5]
```diff
@@ -139,6 +139,7 @@
 
     <av:DataTemplate x:Key="ReferenceTextEditTemplate">
         <dxe:TextEdit Name="PART_Editor"
+                      FontWeight="Bold"
```
Comment (lxatrhea) [inline CDP4CommonView/Resources/ValueSetTemplates.xaml]: Was this change intentional? If so, OK!

File: EngineeringModel/Views/Dialogs/ParameterSubscriptionDialog.xaml (hunk 4 above)
Comment (kamilatrhea) [reply EngineeringModel/Views/Dialogs/ParameterSubscriptionDialog.xaml]: No, my mistake

File: CDP4CommonView/Resources/ValueSetTemplates.xaml (hunk 5 above)
Comment (kamilatrhea) [reply CDP4CommonView/Resources/ValueSetTemplates.xaml]: No, my mistake, I fixed this.
